core/services/ardupilot_manager/firmware_download/FirmwareDownload.py

```diff
@@ -219,24 +250,34 @@ def download(self, vehicle: Vehicle, platform: Platform, version: str = "") -> O
                 if not newest_version or Version(newest_version) < Version(semver_version):
                     newest_version = semver_version
             if not newest_version:
+                logger.error(f"No firmware versions found for this configuration ({vehicle} and {platform}).")
                 return None
             version = f"STABLE-{newest_version}"
 
+        firmware_format = FirmwareDownload._supported_firmware_formats[platform]
+
         items = self._find_version_item(
             vehicletype=vehicle.value,
             platform=platform.value,
             mav_firmware_version_type=version,
-            format=FirmwareDownload._supported_firmware_format,
+            format=firmware_format,
         )
 
         if len(items) != 1:
-            logger.error(f"Invalid number of candidates to download: {len(items)}")
+            logger.error(f"Invalid number of candidates to download ({len(items)}): {items}")
             return None
 
         item = items[0]
+        logger.debug(f"Downloading following firmware: {item}")
         path = FirmwareDownload._download(item["url"])
-        if not path or not FirmwareDownload._validate_firmware(path):
-            logger.error("Unable to validate firmware file.")
-            return None
+        if firmware_format == FirmwareFormat.ELF:
+            # Make the binary executable
+            os.chmod(str(path), stat.S_IXOTH)
+
+        # TODO: Reincorporate firmware validation for SITL once Ardupilot fixes non-zero return on --help (PR #17573)
```

Comment:
That got merged to master. we could use ardusub master(dev) for sitl. I have mixed feeling about it though.
What do you think @patrickelectric ?

Reply:
But is the binary already deployed? We use the manifest to download them, so it need to be deployed to work.

Reply:
last build was today
https://firmware.ardupilot.org/Sub/latest/SITL_x86_64_linux_gnu/

Reply:
Perfect!
Let me try it and I update the PR.

Reply:
@Williangalvani can we cherry pick it to the ardusub version of navigator ?
If its master we are ok.

Reply:
Ok, so the latest version is fixed, but as it's not classified as stable yet, we are ignoring it.

Reply:
Yeah. Let's use master for now. both this patch and proper navigator support are on master only.

Reply:
Done.